SPEC.md

```diff
@@ -111,6 +111,11 @@ Headers that start with the `Nexus-Callback-` prefix are expected to be attached
 the handler. The callback request must strip away the `Nexus-Callback-` prefix. E.g if a Start Operation request
 includes a `Nexus-Callback-Token: some-token` header, the callback request would include a `Token: some-token` header.
 
+If a callback query parameter is provided, the `Nexus-Callback-Token` header is **REQUIRED**. It MUST contain a
```

Comment:
Am curious, why do we require this? Shouldn't it be caller choice on whether they want to provide something to be provided back to them? Granted Temporal caller will always set it, but why does everyone have to? Is there something server side that requires it?

Reply:
> but felt like just Token is a bit too short and we would want something more descriptive for a header that is well known.

Not following this exactly. The server on the handler side doesn't choice right since we just take all `Nexus-Callback-` headers and remove the prefix? Or are we saying the handler server understands token (it makes sense the caller server understands token, it creates/sets it and receives it).